tools/tools.go

```diff
@@ -0,0 +1,178 @@
+package tools
+
+import (
+	"errors"
+	"strings"
+	gotmpl "text/template"
+
+	"github.com/ollama/ollama/api"
+	"github.com/ollama/ollama/template"
+)
+
+// Sentinel errors for parsing states
+var (
+	ErrPartialPrefix = errors.New("partial prefix detected")
+
+	ErrPrefixNotFound = errors.New("prefix not found")
```

Comment:
Is this used?

Reply:
Going to remove these actually - don't need sentinel errors anymore